src/posix/platform/uart.cpp

```diff
@@ -299,10 +299,27 @@ void platformUartProcess(const fd_set *aReadFdSet, const fd_set *aWriteFdSet, co
     {
 #if OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
         // Don't die on SIGPIPE
+#if !defined(MSG_NOSIGNAL)
+        // some platforms (mac OS, Solaris) don't have MSG_NOSIGNAL
+        // SOME of those (mac OS, but NOT Solaris) support SO_NOSIGPIPE
+        // if we have SO_NOSIGPIPE, then set it.  otherwise, we're going
+        // to simply ignore it.
+#if defined(SO_NOSIGPIPE)
+        int err;
+        int flag;
+
+        flag = 1;
+        err  = setsockopt(fd, SOL_SOCKET, SO_NOSIGPIPE, &flag, sizeof(flag));
+        VerifyOrDie(err == 0, OT_EXIT_ERROR_ERRNO);
+#else
+#warning "no support for MSG_NOSIGNAL or SO_NOSIGPIPE"
+#endif
+#define MSG_NOSIGNAL 0
```

Comment:
If we have neither `MSG_NOSIGNAL` nor `SO_NOSIGPIPE`, shall we fallback to `write` rather than manually define `MSG_NOSIGNAL`?

Reply:
Not sure if there's an advantage -- send with flags set to zero is basically the same as write.  Either way, that particular platform is doing something different than ones we normally use (hence the warning so the developer notices).  I don't have a strong preference.